### PR TITLE
[PATCH v2] Improve global init API specification

### PR DESCRIPTION
**Pull request by psavol**
Improve API specification text to be more precise e.g. about which functions may be used before init_global call. API functionality is not changed.

Also fix/improve init validation test suite. Fixed also a bug found in buffer validation test status output.

### CODE REVIEW

File: include/odp/api/spec/init.h [hunk 1]
```diff
@@ -234,13 +225,25 @@ void odp_init_param_init(odp_init_t *param);
...
  * Global ODP initialization
  *
- * This function must be called once (per instance) before calling any other
- * ODP API functions. A successful call creates a new ODP instance into the
- * system and outputs a handle for it. The handle is used in other calls
- * (e.g. odp_init_local()) as a reference to the instance. When user provides
- * configuration parameters, the platform may configure and optimize the
- * instance to match user requirements. A failure is returned if requirements
- * cannot be met.
+ * An ODP instance is created with a odp_init_global() call. By default, each
```
Comment (Bill-Fischofer-Personal) [inline include/odp/api/spec/init.h]: grammar: `with an odp_init_global() call.`

File: include/odp/api/spec/init.h [hunk 2]
```diff
@@ -260,55 +263,33 @@ void odp_init_param_init(odp_init_t *param);
...
  *
  * All threads must call this function before calling any other ODP API
- * functions. The instance parameter specifies which ODP instance the thread
- * joins. A thread may be simultaneously part of single ODP instance only.
+ * functions. See odp_init_global() documentation for exceptions to this rule.
+ * Global initialization (odp_init_global()) must have completed prior calling
+ * this function.
+ *
+ * The instance parameter specifies which ODP instance the thread joins.
+ * A thread may be simultaneously part of single ODP instance only. The thread
```
Comment (Bill-Fischofer-Personal) [inline include/odp/api/spec/init.h]: `simultaneously` seems somewhat awkward here. Perhaps `A thread may only join a single ODP instance at a time.`?

File: include/odp/api/spec/init.h [hunk 3]
```diff
@@ -317,24 +298,40 @@ int odp_init_local(odp_instance_t instance, odp_thread_type_t thr_type);
  *
  * This function is the second to final ODP call made when terminating
  * an ODP application in a controlled way. It cannot handle exceptional
- * circumstances. In general it calls the API modules per thread terminate
- * functions in the reverse order to that which the module init functions were
- * called during odp_init_local().
+ * circumstances. It is recommended that application frees all ODP resources
```
Comment (Bill-Fischofer-Personal) [inline include/odp/api/spec/init.h]: grammar: `that applications free all...`

**Comment by matiaselo** [comment]
@Bill-Fischofer-Personal, Petri is on vacation for this week and most of the next one, but the issues you pointed out should be fixed in the next revision.

@mvl-skardach could you also review these modifications when you have time?

File: include/odp/api/spec/init.h (hunk 1 above)
Comment (psavol) [reply include/odp/api/spec/init.h]: Fixed.

File: include/odp/api/spec/init.h (hunk 2 above)
Comment (psavol) [reply include/odp/api/spec/init.h]: Fixed.

File: include/odp/api/spec/init.h (hunk 3 above)
Comment (psavol) [reply include/odp/api/spec/init.h]: Changed to this: "It is recommended that all ODP resources are freed before the last thread (of the instance) calls this function."

Plural didn't sound good to me as the sentence before is in singular: "...terminating
 an ODP application in a controlled way."

**Comment by psavol** [comment]
Bill, thank you for the review. Which "reviewed-by" tag you'd like to be added into the git log?

**Comment by Bill-Fischofer-Personal** [comment]
@psavol As I'm not longer formally affiliated with the project, I only offer comments rather than reviews. But if you wish feel free to add:

Reviewed-by: Bill Fischofer <billf@me.com>

**Comment by psavol** [comment]
v2: Rebased, fixed typos and added review tags

**Pull request merged**